src/solver/optim/ortoolsImpl/mipConstraint.cpp

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2007-2024, RTE (https://www.rte-france.com)
+ * See AUTHORS.txt
+ * SPDX-License-Identifier: MPL-2.0
+ * This file is part of Antares-Simulator,
+ * Adequacy and Performance assessment for interconnected energy networks.
+ *
+ * Antares_Simulator is free software: you can redistribute it and/or modify
+ * it under the terms of the Mozilla Public Licence 2.0 as published by
+ * the Mozilla Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Antares_Simulator is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Mozilla Public Licence 2.0 for more details.
+ *
+ * You should have received a copy of the Mozilla Public Licence 2.0
+ * along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
+ */
+
+#include <antares/logs/logs.h>
+#include <antares/solver/optim/ortoolsImpl/mipConstraint.h>
+#include <antares/solver/optim/ortoolsImpl/mipVariable.h>
+
+namespace Antares::Solver::Optim::OrtoolsImpl
+{
+
+OrtoolsMipConstraint::OrtoolsMipConstraint(operations_research::MPConstraint* mpConstraint):
+    mpConstraint_(mpConstraint)
+{
+}
+
+void OrtoolsMipConstraint::setLb(double lb)
+{
+    mpConstraint_->SetLB(lb);
+}
+
+void OrtoolsMipConstraint::setUb(double ub)
+{
+    mpConstraint_->SetUB(ub);
+}
+
+void OrtoolsMipConstraint::setBounds(double lb, double ub)
+{
+    mpConstraint_->SetBounds(lb, ub);
+}
+
+double OrtoolsMipConstraint::getLb()
+{
+    return mpConstraint_->lb();
+}
+
+double OrtoolsMipConstraint::getUb()
+{
+    return mpConstraint_->ub();
+}
+
+void OrtoolsMipConstraint::setCoefficient(Api::MipVariable* var, double coefficient)
+{
+    auto* mpvar = dynamic_cast<OrtoolsMipVariable*>(var);
```

Comment:
I think we can avoid dynamic cast by storing in the OrToolsMipConstraint a pointer to the OrToolsLinearProblem, and making the list of variables in OrtoolsLinearProblem accessible to OrtoolsMipConstraint.

then we'd have something like :
auto* mpvar = ortoolsLinearProblem.getMipVariable(var.getName());

(you would need to change the return value of OrtoolsLinearProblem::getMipVariable to a OrtoolsMipVariable, which is OK from a polymorphism point-of-view)

but I'm not really sure it's worth the hassle. What do you think?

Reply:
If we do what you suggest, would not we have a performance issue ?
I mean : we need to search for a var in a list for each call to **setCoefficient(...)**, right ?